components/ws-manager-bridge/src/bridge.ts

```diff
@@ -508,36 +514,33 @@ export class WorkspaceManagerBridge implements Disposable {
             for (const [instanceId, ri] of runningInstancesIdx.entries()) {
                 const instance = ri.latestInstance;
                 const phase = instance.status.phase;
-                if (phase !== "running") {
-                    // This below if block is to validate the planned fix
-                    if (
-                        phase === "pending" ||
-                        phase === "creating" ||
-                        phase === "initializing" ||
-                        (phase === "stopping" &&
-                            instance.stoppingTime &&
-                            durationLongerThanSeconds(Date.parse(instance.stoppingTime), 10))
-                    ) {
-                        log.info(
-                            { instanceId, workspaceId: instance.workspaceId },
-                            "Logging to validate #12902. Should mark as stopped in database.",
-                            { installation },
-                            { phase },
-                        );
-                    }
-                    log.debug({ instanceId }, "Skipping instance", {
-                        phase: instance.status.phase,
-                        creationTime: instance.creationTime,
-                        region: instance.region,
-                    });
+
+                // When ws-manager is not aware of the following instances outside of the timeout duration,
+                // they should be marked as stopped.
+                // pending states timeout is 1 hour after creationTime.
+                // stopping states timeout is 1 hour after stoppingTime.
+                if (
+                    (phase === "pending" &&
+                        durationLongerThanSeconds(Date.parse(instance.creationTime), maxInstanceDuration)) ||
```

Comment:
Would be great to have separate timeouts per phase. This is something that does change, even if it's very rare.

Reply:
Aha, sure. So for now they would be something like `maxPendingInstanceDuration` and `maxStoppingInstanceDuration`? And both would be 3600 seconds for now.

Reply:
Why not continue the pattern [here](https://github.com/gitpod-io/gitpod/pull/13350/files#diff-679f493f2181427cc1a110c353e46be2ac044f9f0648b7b89a84b0d0958101d6R31-R33): `pendingPhaseSeconds` and `stoppingPhaseSeconds`? Not caring so much about the particular names, more about consistency. :lotus_position: